### ISSUE TITLE
[3.1.6] Any error which is not a Syntax Error is said to be on line 1

### ISSUE THREAD

**Issue by JJFlash-IT** [neilsf/xc-basic3#234]
>XC=BASIC compiler version v3.1.6 (Jul 18 2023)
>**(Windows 7/10, 64 bit)**

First of all, thank you so much for fixing the NEXT-FOR problem and so quickly!

Unfortunately, there's still the "line 1" problem I had mentioned in my previous issue.

Again, for your convenience, I'm uploading the source code of Advent 101 here:
[Advent_101.zip](https://github.com/neilsf/xc-basic3/files/12091168/Advent_101.zip)

To reproduce the problem:
- Try to edit line **92** of the _advent101.bas_ file, changing `declare` to `dclare`. Once you compile it, the compiler says `advent101.bas:92.7: syntax error near 'sub strngP' in file advent101.bas in line 92` . And that's fine, expected behaviour;
- Try to edit line **212** of the _advent101.bas_ file, changing `call strngPrnt` to `call blah`. Once you compile it, the compiler says `advent101.bas:1.0: ERROR: Unknown identifier: blah(LNG_STRNG,@fixed_str_gamename,32,0,%11000000)`, which is **not** fine, it should have said "212.0" at least, instead of "1.0".

The same happens for any error which is not a Syntax Error.

My program INCLUDEs quite a few files, but it seems that `INCLUDE` does not interfere: I tried commenting out all the INCLUDEs at the beginning of _advent101.bas_, it still says "1.0".

I hope a solution can be found, as it can be quite frustrating not to know where a problem can be, while developing a medium if not large sized program.

**Comment by neilsf** [comment]
Hi @JJFlash-IT 
It looks like an issue on Windows only. I added a fix that I can't test right now, would you be able to check out the develop branch and run a quick test? Many thanks.

**Comment by JJFlash-IT** [comment]
Hey! Sorry for getting back to you so late, I was on a short vacation and also had to figure out how to use DMD on Windows (I'm totally clumsy with modern development tools and solutions) to compile the source on the "develop" branch.

I hope I did everything right, anyway the fix seems to work great: I tried a number of changes in the main _advent101.bas_ file and also in a couple of the INCLUDEd files.

I even tried to use different newline sequences (I went from CRLF to LF only and then back) in the _advent101.bas_ file, and also in the INCLUDEd files (mixing various combinations among the files), and the compiler always gave me the correct line number where I put errors on purpose.

I suppose I'll repeat those tests once this fix goes into the main branch / actual release, anyway the issue seems to be fixed. Thank you once again for quickly addressing it!
